controllers/nutanixcluster_controller.go

```diff
@@ -250,8 +263,7 @@ func (r *NutanixClusterReconciler) reconcileNormal(rctx *nctx.ClusterContext) (r
 
 	err := r.reconcileCategories(rctx)
 	if err != nil {
-		errorMsg := fmt.Errorf("failed to reconcile categories for cluster %s: %v", rctx.Cluster.Name, err)
-		rctx.SetFailureStatus(capierrors.CreateClusterError, errorMsg)
+		// Don't return fatal error but keep retrying until categories are created.
 		return reconcile.Result{}, err
```

Comment:
we may need to requeue here in order for the retry to work else it will assume success in reconciliation